website_sale_close/tests/test_website_sale_close.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2019-Today Coop IT Easy SCRLfs
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import HttpCase
+
+
+class WebsiteSaleCloseCase(HttpCase):
+
+    def test_01_check_e_commerce_open(self):
+        website = self.env["website"].browse(1)
+        website.is_ecommerce_open = True
+        result = self.url_open("/shop")
+        self.assertNotIn(website.text_closed_ecommerce, result.text)
+
+    def test_02_check_e_commerce_close(self):
+        website = self.env["website"].browse(1)
+        website.is_ecommerce_open = False
+        result = self.url_open("/shop")
+        self.assertIn(website.text_closed_ecommerce, result.text)
```

Comment:
Is this working? I thought that the `website.is_ecommerce_open` would be in the current transaction while the http request in `url_open` would reach the server in another transaction.

Reply:
It seams to work. With debugging breakpoint, the returned content of shop differ accordingly to `website.is_ecommerce_open`. Note that I use a build in odoo function `url_open`.

Reply:
ha, interesting